### PR TITLE
Baal d config1

### PR DESCRIPTION
**Pull request by sparkie3**
Added two configuration options:

Diablo.ClearPath
Baal.ClearPath

These will clear the path on the way to the Chaos Sanctuary and Baal respectively. First pull request, but code formatting conventions were followed, though I did use a case statement rather than if statements because it seems to make the code neater and easier to read in this case.

### CODE REVIEW

File: d2bs/kolbot/libs/bots/Baal.js [hunk 1]
```diff
@@ -203,9 +203,18 @@ function Baal() {
 	if (me.area !== 129) {
 		Pather.useWaypoint(129);
 	}
-
-	if (!Pather.moveToExit([130, 131], true)) {
-		throw new Error("Failed to move to Throne of Destruction.");
+//Check Config.Baal.ClearPath option for true or false and clear the path if walking.
```
Comment (noah-) [inline d2bs/kolbot/libs/bots/Baal.js]: remove the comment, code is self explanatory change to this:
```
if (!Pather.moveToExit([130, 131], true, Config.Baal.ClearPath)) {
	throw new Error("Failed to move to Throne of Destruction.");
}
```

File: d2bs/kolbot/libs/bots/Diablo.js [hunk 2]
```diff
@@ -429,16 +429,33 @@ function Diablo() {
 	if (me.area !== 107) {
 		Pather.useWaypoint(107);
 	}
-
-	if (!Pather.moveTo(7790, 5544)) {
-		throw new Error("Failed to move to Chaos Sanctuary");
+//Check Config.Diablo.ClearPath option for true or false and clear the path if walking.
```
Comment (noah-) [inline d2bs/kolbot/libs/bots/Diablo.js]: don't think the say is necessary, the throw error string you have is wrong, logic same as above change to this:
```
if (!Pather.moveTo(7790, 5544, 5, Config.Diablo.ClearPath)) {
	throw new Error("Failed to move to Chaos Sanctuary");
}
```

File: d2bs/kolbot/libs/bots/Diablo.js [hunk 3]
```diff
@@ -429,16 +429,33 @@ function Diablo() {
...
 	}
+	
 
 	this.initLayout();
 
 	if (Config.Diablo.Entrance) {
 		Attack.clear(30, 0, false, this.sort);
-		Pather.moveTo(7790, 5544);
+			
+		if (Config.Diablo.ClearPath) {
```
Comment (noah-) [inline d2bs/kolbot/libs/bots/Diablo.js]: change logic same as above:
`Pather.moveTo(7790, 5544, 3, Config.Diablo.ClearPath);`

File: d2bs/kolbot/libs/config/Amazon.js [hunk 4]
```diff
@@ -86,6 +86,7 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
+		Config.Diablo.ClearPath = false; //Clear path if walking to Diablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Amazon.js]: add space after //

File: d2bs/kolbot/libs/config/Assassin.js [hunk 5]
```diff
@@ -86,6 +86,7 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
+		Config.Diablo.ClearPath = false; //Clear path if walking to Diablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Assassin.js]: add space after //

File: d2bs/kolbot/libs/config/Barbarian.js [hunk 6]
```diff
@@ -86,6 +86,7 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
+		Config.Diablo.ClearPath = false; //Clear path if walking to Diablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Barbarian.js]: add space after //

File: d2bs/kolbot/libs/config/Druid.js [hunk 7]
```diff
@@ -86,6 +86,7 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
+		Config.Diablo.ClearPath = false; //Clear path if walking to Diablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Druid.js]: add space after //

File: d2bs/kolbot/libs/config/Paladin.js [hunk 8]
```diff
@@ -86,6 +86,7 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
+		Config.Diablo.ClearPath = false; //Clear path if walking to Diablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Paladin.js]: add space after //

File: d2bs/kolbot/libs/config/Sorceress.js [hunk 9]
```diff
@@ -86,6 +86,7 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
+		Config.Diablo.ClearPath = false; //Clear path if walking to Diablo, regardless of if the character is teleporting or not.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Sorceress.js]: add space after //

**Comment by sparkie3** [comment]
Not sure what happened, I thought I had cleaned that up already. Oh well, first time using git. Anywho, changes are made and pushed to my fork, just not sure how to update this, or if it updates automatically.

**Comment by sparkie3** [comment]
Changes made and pushed. Sorry about the delay

**Comment by noah-** [comment]
Please look at ALL the comments made int he review. I still see various logic issues, indentation, and lack of jslint being run on the changes.

File: d2bs/kolbot/libs/bots/Baal.js [hunk 10]
```diff
@@ -204,11 +204,11 @@ function Baal() {
 		Pather.useWaypoint(129);
 	}
 
-	if (!Pather.moveToExit([130, 131], true)) {
+	if (!Pather.moveToExit([130, 131], true, Config.Baal.Clearpath)) {
 		throw new Error("Failed to move to Throne of Destruction.");
-	}
+		}
```
Comment (noah-) [inline d2bs/kolbot/libs/bots/Baal.js]: Random tab here

File: d2bs/kolbot/libs/bots/Diablo.js [hunk 11]
```diff
@@ -429,28 +429,25 @@ function Diablo() {
...
 		Pather.useWaypoint(107);
 	}
-
-	if (!Pather.moveTo(7790, 5544)) {
-		throw new Error("Failed to move to Chaos Sanctuary");
-	}
+	
+	if (!Pather.moveTo(7790, 5544, 5, Config.Diablo.ClearPath)) {
+		throw new Error("Failed to move to The Chaos Sanctuary.");
+		}
```
Comment (noah-) [inline d2bs/kolbot/libs/bots/Diablo.js]: Line 435 same formatting issue here, extra spaces

File: d2bs/kolbot/libs/bots/Diablo.js [hunk 12]
```diff
@@ -429,28 +429,25 @@ function Diablo() {
...
-		}
-
+		Pather.moveTo(7790, 5544, 3, Config.Diablo.ClearPath);
 		Pather.moveTo(7790, 5544);
 		Precast.doPrecast(true);
 		Attack.clear(30, 0, false, this.sort);
 		this.followPath(this.entranceToStar);
-	} else {
-		Pather.moveTo(7774, 5305);
+	}
```
Comment (noah-) [inline d2bs/kolbot/libs/bots/Diablo.js]: between lines 447 and 449 this if/else statement is not formatted correctly

File: d2bs/kolbot/libs/common/Config.js [hunk 13]
```diff
@@ -343,7 +343,8 @@ var Config = {
 		KillBaal: false,
 		HotTPMessage: "Hot TP!",
 		SafeTPMessage: "Safe TP!",
-		BaalMessage: "Baal!"
+		BaalMessage: "Baal!",
+		ClearPath: false
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Config.js]: move ClearPath: as the first option

File: d2bs/kolbot/libs/common/Config.js [hunk 14]
```diff
@@ -378,7 +379,8 @@ var Config = {
 		SealWarning: "Leave the seals alone!",
 		EntranceTP: "Entrance TP up",
 		StarTP: "Star TP up",
-		DiabloMsg: "Diablo"
+		DiabloMsg: "Diablo",
+		ClearPath: false
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Config.js]: move ClearPath: as the first option

File: d2bs/kolbot/libs/config/Amazon.js [hunk 15]
```diff
@@ -86,6 +86,7 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
+		Config.Diablo.ClearPath = false; // Clear path if walking to Diablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Amazon.js]: move ClearPath as the first option

File: d2bs/kolbot/libs/config/Amazon.js [hunk 16]
```diff
@@ -117,6 +118,7 @@ function LoadConfig() {
 		Config.Baal.SoulQuit = false; // End script if Souls (Undead Soul Killers) are found.
 		Config.Baal.DollQuit = false; // End script if Dolls (Undead Stigyan Dolls) are found.
 		Config.Baal.KillBaal = true; // Kill Baal. Leaves game after wave 5 if false.
+		Config.Baal.ClearPath = false; // Clear the path to Baal if walking.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Amazon.js]: move ClearPath as the first option

File: d2bs/kolbot/libs/config/Assassin.js [hunk 17]
```diff
@@ -86,6 +86,7 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
+		Config.Diablo.ClearPath = false; // Clear path if walking to Diablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Assassin.js]: move ClearPath as the first option

File: d2bs/kolbot/libs/config/Assassin.js [hunk 18]
```diff
@@ -117,6 +118,7 @@ function LoadConfig() {
 		Config.Baal.SoulQuit = false; // End script if Souls (Undead Soul Killers) are found.
 		Config.Baal.DollQuit = false; // End script if Dolls (Undead Stigyan Dolls) are found.
 		Config.Baal.KillBaal = true; // Kill Baal. Leaves game after wave 5 if false.
+		Config.Baal.ClearPath = false; // Clear the path to Baal if walking.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Assassin.js]: move ClearPath as the first option

File: d2bs/kolbot/libs/config/Barbarian.js [hunk 19]
```diff
@@ -86,6 +86,7 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
+		Config.Diablo.ClearPath = false; // Clear path if walking to Diablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Barbarian.js]: move ClearPath as the first option

File: d2bs/kolbot/libs/config/Barbarian.js [hunk 20]
```diff
@@ -117,6 +118,7 @@ function LoadConfig() {
 		Config.Baal.SoulQuit = false; // End script if Souls (Undead Soul Killers) are found.
 		Config.Baal.DollQuit = false; // End script if Dolls (Undead Stigyan Dolls) are found.
 		Config.Baal.KillBaal = true; // Kill Baal. Leaves game after wave 5 if false.
+		Config.Baal.ClearPath = false; // Clear the path to Baal if walking.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Barbarian.js]: move ClearPath as the first option

File: d2bs/kolbot/libs/config/Druid.js [hunk 21]
```diff
@@ -86,6 +86,7 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
+		Config.Diablo.ClearPath = false; // Clear path if walking to Diablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Druid.js]: move ClearPath as the first option

File: d2bs/kolbot/libs/config/Druid.js [hunk 22]
```diff
@@ -117,6 +118,7 @@ function LoadConfig() {
 		Config.Baal.SoulQuit = false; // End script if Souls (Undead Soul Killers) are found.
 		Config.Baal.DollQuit = false; // End script if Dolls (Undead Stigyan Dolls) are found.
 		Config.Baal.KillBaal = true; // Kill Baal. Leaves game after wave 5 if false.
+		Config.Baal.ClearPath = false; // Clear the path to Baal if walking.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Druid.js]: move ClearPath as the first option

File: d2bs/kolbot/libs/config/Necromancer.js [hunk 23]
```diff
@@ -86,7 +86,8 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
-	Scripts.SealLeader = false; // Clear a safe spot around seals and invite leechers in. Leechers should run SealLeecher script. Don't run with Diablo or FastDiablo.
+		Config.Diablo.ClearPath = false; // Clear path if walking to Diablo.
+		Scripts.SealLeader = false; // Clear a safe spot around seals and invite leechers in. Leechers should run SealLeecher script. Don't run with Diablo or FastDiablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Necromancer.js]: Remove indentation before Scripts.SealLeader

File: d2bs/kolbot/libs/config/Necromancer.js [hunk 24]
```diff
@@ -86,7 +86,8 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
-	Scripts.SealLeader = false; // Clear a safe spot around seals and invite leechers in. Leechers should run SealLeecher script. Don't run with Diablo or FastDiablo.
+		Config.Diablo.ClearPath = false; // Clear path if walking to Diablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Necromancer.js]: move ClearPath as the first option

File: d2bs/kolbot/libs/config/Necromancer.js [hunk 25]
```diff
@@ -117,6 +118,7 @@ function LoadConfig() {
 		Config.Baal.SoulQuit = false; // End script if Souls (Undead Soul Killers) are found.
 		Config.Baal.DollQuit = false; // End script if Dolls (Undead Stigyan Dolls) are found.
 		Config.Baal.KillBaal = true; // Kill Baal. Leaves game after wave 5 if false.
+		Config.Baal.ClearPath = false; // Clear the path to Baal if walking.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Necromancer.js]: move ClearPath as the first option

File: d2bs/kolbot/libs/config/Paladin.js [hunk 26]
```diff
@@ -86,6 +86,7 @@ function LoadConfig() {
 		Config.Diablo.EntranceTP = "Entrance TP up";
 		Config.Diablo.StarTP = "Star TP up";
 		Config.Diablo.DiabloMsg = "Diablo";
+		Config.Diablo.ClearPath = false; // Clear path if walking to Diablo.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Paladin.js]: move ClearPath as the first option

File: d2bs/kolbot/libs/config/Paladin.js [hunk 27]
```diff
@@ -117,6 +118,7 @@ function LoadConfig() {
 		Config.Baal.SoulQuit = false; // End script if Souls (Undead Soul Killers) are found.
 		Config.Baal.DollQuit = false; // End script if Dolls (Undead Stigyan Dolls) are found.
 		Config.Baal.KillBaal = true; // Kill Baal. Leaves game after wave 5 if false.
+		Config.Baal.ClearPath = false; // Clear the path to Baal if walking.
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Paladin.js]: move ClearPath as the first option

**Comment by noah-** [comment]
Please rebase and resubmit this PR.

**Pull request closed without merging**